napari_aicssegmentation/controller/batch_processing_controller.py

```diff
@@ -13,8 +17,53 @@ def __init__(self, application: IApplication, workflow_engine: WorkflowEngine):
         self._workflow_engine = workflow_engine
         self._view = BatchProcessingView(self)
 
+        # Should these go into a model?
+        self.input_folder: Path = None
+        self.output_folder = None
+        self.selected_index = None
+        self.workflow_config = None
+
     def index(self):
         self.load_view(self._view)
 
     def run_batch(self):
-        return super().run_batch()
+        # workflow = self.get_batch_workflow()
+        # workflow.process_all()
+        self.worker = create_worker(self.run_batch_async)
+        self.worker.start()
+
+    def run_batch_async(self):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            workflow = self.get_batch_workflow()
+            workflow.process_all()
+
+    def ready_to_process(self):
+        if not self.workflow_config:
+            return False
+        elif not self.input_folder:
+            return False
+        elif not self.output_folder:
+            return False
+        else:
+            return True
+
+    def select_config(self, selected_config):
+        self.workflow_config = Path(selected_config)
+        if self.ready_to_process():
+            self._view.update_button(enabled=True)
+
+    def select_input_folder(self, input_folder):
+        self.input_folder = Path(input_folder)
+        if self.ready_to_process():
+            self._view.update_button(enabled=True)
```

Comment:
I think your current code only enables the button if it can be enabled, but doesn't disable it if the user does something funky and removes some inputs. Not sure it's entirely possible through the UI right now, but it could be, or might be in the future if the UI changes a bit.
I would just replace this
```
        if self.ready_to_process():
            self._view.update_button(enabled=True)
```
with this:
```
self._view.update_button(enabled=self.ready_to_process())
```